### ISSUE TITLE
Workers gain speed spell too fast

### ISSUE THREAD

**Issue by Loobinex** [OpenDungeons/OpenDungeons#1168]
Workers get the speed spell at level 8, which they reach really soon after you created them. Because of this, basically all your workers will have the speed spell active and it effectively becomes the norm.

What would be more fun, is that having speedy workers is a true reward for having kept your workers alive for a long time. I would suggest that you must have a worker active for something like 20 or 30 minutes before it gains the speed spell.

This can either be accomplished by raising the level it acquires this skill and/or by decreasing how fast it gains levels.
As I feel the speed of the workers currently is just fine, I also suggest to boost the base speed of the workers to match the current speed of the worker when the speed spell is active.

**Comment by hwoarangmy** [comment]
Yes, I also think they level too fast. That's part of the balancing we need for the game. The easiest would be to change the XP table in config/creatures.cfg. If you want to make some tries and come up with something, that would be great as you probably play more the game as most of us ^^

> I would suggest that you must have a worker active for something like 20 or 30 minutes before it gains the speed spell

Wow, you're hitting hard. We aim for an average multiplayer game of  minutes so I guess it is too much. But 5 minutes seems pretty fair to me.

**Comment by Loobinex** [comment]
I've yet to finish a single level. Last time I died to a bunch of wizards(weeks ago), now I'm trying a new level with modified stats for the wizard but as there's no game map a single level is taking forever. They are so big, with so much gold and no indication of where the enemy is that it takes hours and becomes dull.

Having a multiplayer level just take minutes is silly though. It's a dungeon-sim game not a FPS where you play multiple quick matches in a row. Unless one player rushes and multiplayer maps should contain some creatures to make it risky for the player to dig straight to the enemy before he has trained up.

However, on very quick games I don't think it is a problem if you don't get speed-workers. As a general rule of thumb workers should never gain speed before you've had contact with enemy creatures as the speed should be a reward for keeping them alive a long time and that is not the case if they simply haven't gotten a chance to die yet.

**Comment by Loobinex** [comment]
I've looked into the exp numbers of the worker a bit, and he gains massive amounts of XP. Hardly anything when fortifying walls, a bit more when claiming, lots on digging and lots more on mining. I think fortifying should net more XP, and mining gold may be reduced a bit.

If a worker mines out a single 25 tile room, claims and fortifies it and mines 9 gold blocks he has acquired roughly 1000xp, most of it from the gold . It takes 218 xp to get to level 8, so with this very basic task you can get 5 imps to get speed before you've even properly started.

I suggest to try to set the XP values of the workers to:

```
    [XP]
    # 2-10
    300 400 475 550 625 700 775 850 925
    # 11-20
    1000 1025 1050 1075 1100 1125 1150 1175 1200 1225
    # 21-30
    1250 1275 1300 1325 1350 1375 1400 1425 1450 1475
    [/XP]
```

And give out the speed skill on lvl 10:

```
[CreatureSkills]
    # SkillName CooldownNbTurns WarmupNbTurns   LevelMin    PhyAtk  PhyAtkPerLvl    MagAtk  MagAtkPerLvl    EleAtk  EleAtkPerLvl
    Melee   2   1   1   2   0.1 0   0   0   0
    # SkillName CooldownNbTurns WarmupNbTurns   LevelMin    EffectDuration  EffectValue
    HasteSelf   30  0   10  10  1.2
    [/CreatureSkills]
```

**Comment by hwoarangmy** [comment]
I'm ok with that and I will give a try

@OpenDungeons/developers WDYT ?

**Comment by Danimal696** [comment]
its true that rigth now Xp and levels  gains are quite messed up, i agree to any chance that bring some balance

**Comment by Bertram25** [comment]
Why not? :)
